### PR TITLE
use local variable for max_amount property

### PR DESCRIPTION
**Pull request by daliwei**

### CODE REVIEW

**Comment by daliwei** [comment]
The second commit updates the rake version in gemspec because the latest versions > 11.0.0 released on March 9th, 2016 are no more compatible.

**Comment by pierre** [comment]
> The second commit updates the rake version in gemspec because the latest versions > 11.0.0 released on March 9th, 2016 are no more compatible.

I'm not sure I follow. You are now constraining the version to be `10.4.2 =< version < 10.5`. How about:

```
s.add_dependency 'rake', '>= 10.0.0', '< 11.0.0'
```

Alternatively, we could investigate what would it take to upgrade (and/or file an issue for it).

**Comment by daliwei** [comment]
Yea.  `s.add_dependency 'rake', '>= 10.0.0', '< 11.0.0'` is better.

If the version is >= 11.0.0, when you run rake commands (e.g., rake killbill:clean), it will give the following error. Looks like the rspec gem is using "last_comment" which has been removed in rake 11.0.1

image

**Comment by pierre** [comment]
Ah, yup: http://stackoverflow.com/questions/35893584/nomethoderror-undefined-method-last-comment-after-upgrading-to-rake-11/35893941.

File: Gemfile.lock [hunk 1]
```diff
@@ -19,13 +19,13 @@ PATH
 GEM
   remote: https://rubygems.org/
   specs:
-    actionpack (4.1.14.1)
-      actionview (= 4.1.14.1)
-      activesupport (= 4.1.14.1)
+    actionpack (4.1.15)
```
Comment (pierre) [inline Gemfile.lock]: This is a lot of updates, did you run the integration tests against a PayPal sandbox (`rake test:remote:spec`)? I'm a bit nervous about side effects.

Alternatively, we could tackle these upgrades separately if we need/want them.

**Comment by daliwei** [comment]
Ok. I have removed these commits from the pull requests. We can tackle these updates separately. BTW, I did not know the plugin itself can run tests against PayPal. Will try that!

**Comment by pierre** [comment]
> Ok. I have removed these commits from the pull requests. We can tackle these updates separately.

:+1: 

> BTW, I did not know the plugin itself can run tests against PayPal. Will try that!

Yup, there are unit and integration tests! :+1: for adding more :)

**Comment by pierre** [comment]
Note that the `rake` changes are also breaking [KPM](https://travis-ci.org/killbill/killbill-cloud/jobs/115201478).

**Comment by daliwei** [comment]
ack. Another maybe irrelevant question: why KPM excludes gemfile.lock in the repo?

**Comment by pierre** [comment]
It's because we distribute KPM as a gem. The one for the [KPM plugin](https://github.com/killbill/killbill-kpm-plugin) however is checked-in.

Some context: http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/.

**Comment by daliwei** [comment]
Gotcha, thanks!

**Pull request merged**